test/manual/resource_management.test.ts

```diff
@@ -0,0 +1,89 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+import { AbstractCursor, ChangeStream, ClientSession, MongoClient } from '../mongodb';
+
+describe('Explicit Resource Management Tests', function () {
+  let client: MongoClient;
+
+  afterEach(async function () {
+    await client?.close();
+  });
+
+  describe('Symbol.asyncDispose defined', function () {
+    beforeEach(function () {
+      if (!('asyncDispose' in Symbol)) {
+        this.currentTest.skipReason = 'Test must run with asyncDispose available.';
+        this.skip();
+      }
+    });
+
+    describe('MongoClient', function () {
+      it('closes the the client', async function () {
```

Comment:
I suggest flattening the tests (getting rid of the class-level describes) and making the `it` blocks read `'calls <method> when invoked on the <class name>'` in order to make it clear what is performing the action under test and when

Reply:
The previous suggestion was just one way of trying to make it more clear what condition is being tested. The `asyncDispose` being defined doesn't in itself cause the MongoClient to close, which is what the current prose suggests, and MongoClient can be closed regardless of whether `asyncDispose` exists. So you need to add something along the lines of `closes <resource> via direct call to Symbol.asyncDispose` in order to fully convey the intent of the test. (In contrast, the symbol not being defined as opposed to not defined IS what is responsible for the absence of the resulting definition in the negative cases, so those correctly convey their intent).